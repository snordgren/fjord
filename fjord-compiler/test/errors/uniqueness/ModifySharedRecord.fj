module errors.uniqueness.ModifySharedRecord


record Shared 
  sharedInt : Int


illegalMutation : &Shared -> Shared
illegalMutation a =
  { a | sharedInt = 3 }
