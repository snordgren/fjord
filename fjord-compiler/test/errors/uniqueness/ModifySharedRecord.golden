test/errors/uniqueness/ModifySharedRecord.fj:10:3:
   |
10 |   { a | sharedInt = 3 }
   |   ^^^^^^^^^^^^^^^^^^^^^^
expression has type ?:Shared, expected 1:Shared
